src/Xamarin.Android.Build.Tasks/Tasks/GetImportedLibraries.cs

```diff
@@ -36,28 +32,33 @@ public class GetImportedLibraries : Task
 
 		public override bool Execute ()
 		{
-			Log.LogDebugMessage ("GetImportedLibraries Task");
-			Log.LogDebugMessage ("  TargetDirectory: {0}", TargetDirectory);
 			if (!Directory.Exists (TargetDirectory)) {
 				Log.LogDebugMessage ("Target directory was not found");
 				return true;
 			}
-			// there could be ./AndroidManifest.xml and bin/AndroidManifest.xml, which will be the same. So, ignore "bin" ones.
+
 			var manifestDocuments = new List<ITaskItem> ();
-			foreach (var f in Directory.EnumerateFiles (TargetDirectory, "AndroidManifest.xml", SearchOption.AllDirectories)) {
-				var directory = Path.GetFileName (Path.GetDirectoryName (f));
-				if (IgnoredManifestDirectories.Contains (directory))
-					continue;
-				manifestDocuments.Add (new TaskItem (f));
+			var nativeLibraries   = new List<ITaskItem> ();
+			var jarFiles          = new List<ITaskItem> ();
+			foreach (var file in Directory.EnumerateFiles (TargetDirectory, "*", SearchOption.AllDirectories)) {
+				var fileName = Path.GetFileName (file);
```

Comment:
Would it be worth adding an early out for files which do not have a `.xml;.so.jar` file extension? 
Not sure what the performance would be if we have say 1 xml file but 500 png files..

Reply:
The `searchPattern` for `EnumerateFiles` doesn't have a syntax for `!` or `|` or something like that.

I can reorganize this `if-else-if` so the `Path.GetFileName` is only called for XML files. I don't think we'll see a huge improvement (probably less than 1ms), but doesn't hurt.